contracts/src/v0.8/keystone/KeystoneForwarder.sol

```diff
@@ -81,17 +79,98 @@ contract KeystoneForwarder is IForwarder, OwnerIsCreator, ITypeAndVersion {
     bool result
   );
 
-  string public constant override typeAndVersion = "KeystoneForwarder 1.0.0";
+  string public constant override typeAndVersion = "Forwarder and Router 1.0.0";
 
-  constructor(address router) OwnerIsCreator() {
-    s_router = router;
+  constructor() OwnerIsCreator() {
+    s_forwarders[address(this)] = true;
   }
 
   uint256 internal constant MAX_ORACLES = 31;
   uint256 internal constant METADATA_LENGTH = 109;
   uint256 internal constant FORWARDER_METADATA_LENGTH = 45;
   uint256 internal constant SIGNATURE_LENGTH = 65;
 
+  // ================================================================
+  // │                          IRouter                             │
```

Comment:
IRouter -> Router? (This is the implementation I believe, not the interface)

Reply:
Will follow-up in rename.